Fix issue with ransackable_attribute not handling symbols correctly

tatematsu-k opened the pull request:
Fix: https://github.com/activerecord-hackery/ransack/issues/1538

If the behavior is as intended, please close this PR.

---

## Problem
When using the ransackable_attributes method in ActiveRecord, passing an array of symbols causes the search conditions for the corresponding attributes to be ignored. However, passing an array of strings works as expected.

### Steps to Reproduce
1: Create a migration for a users table with a name column:
```rb
class CreateUsers < ActiveRecord::Migration[7.1]
  def change
    create_table :users do |t|
      t.string :name

      t.timestamps
    end
  end
end
```

2: Define the User model with ransackable_attributes returning a symbol array:

```rb
class User < ApplicationRecord
  def self.ransackable_attributes(auth_object = nil)
    [:name]
  end
end
```

3: Attempt to perform a Ransack query with a condition on the name attribute:

`User.ransack(name_cont: "hoge").result.to_sql`

## Actual Behavior
The generated SQL query ignores the condition for name:

`SELECT "users".* FROM "users"`

tatematsu-k committed:
Fix issue with ransackable_attribute not handling symbols correctly

gregmolnar committed:
Merge branch 'main' into feature/fix_ransackable_attributes_allow_symbol

scarroll32 committed:
Merge branch 'main' into feature/fix_ransackable_attributes_allow_symbol

tatematsu-k commented:
This PR has been inactive for a while.
Is there anything I can do to help move it forward?

ypoloni commented:
@tatematsu-k  Thanks a lot! That works well for the 'where' search. However, I also need to override Nodes::Sort with the same solution to get Ransack sorting working as well.

Could you update your pull request changing Nodes::sort too?

```
Ransack::Nodes::Sort.class_eval do
  def valid?
    bound? && attr &&  context.ransackable_attribute?(attr_name.split('.').last, context.klassify(parent))
  end
end
```

tatematsu-k committed:
Fix Nodes::Sort to handle ransortable_attributes returning symbols (PR #1539)

- Add Context#ransortable_attribute? with symbol/string normalization
- Change Sort#valid? to use ransortable_attribute? instead of .include?(attr_name)
- Add spec for sort when ransortable_attributes returns symbol array

Addresses comment from ypoloni: sorting was ignored when model used
symbol array for ransortable_attributes (same issue as #1538 for search).

Made-with: Cursor

tatematsu-k committed:
Fix: Separate responsibilities of ransackable_attribute? and ransortable_attribute?

- Remove ransortable_attributes check from ransackable_attribute?
- Explicitly check ransortable_attribute? in attribute_method? and get_parent_and_attribute_name in ActiveRecord context

Made-with: Cursor
